homeassistant/components/philips_js/config_flow.py

```diff
@@ -30,18 +37,19 @@
             CONF_API_VERSION,
             default=1,
         ): vol.In([1, 5, 6]),
+        vol.Required(CONF_USE_HTTPS, default=False): bool,
     }
 )
 
 
 async def _validate_input(
-    hass: core.HomeAssistant, host: str, api_version: int
+    hass: core.HomeAssistant, host: str, api_version: int, secured_transport: bool
 ) -> PhilipsTV:
     """Validate the user input allows us to connect."""
-    hub = PhilipsTV(host, api_version)
+    hub = PhilipsTV(host, api_version, secured_transport=secured_transport)
 
+    await hub.setTransport(secured_transport)
     await hub.getSystem()
-    await hub.setTransport(hub.secured_transport)
```

Comment:
This must remain after getSystem() otherwise auto detect fail. Also can remain as is. Once it get a system structure it is likely correct.

Reply:
Ps. We dont even need to store this setting in the config entry. It will be deduced from the system structure once it has connected once. Its only needed during initial connect.

Reply:
You pass it in the constructor. Thay should be enough?

Reply:
Then what is the point of setting it here again?

Reply:
It how the autodetection works. getSystem will modify the property, but not the setting on how we access.